flake.nix

```diff
@@ -3,180 +3,25 @@
-            modules = [
-              dummy
-              fake-hardware
-              self.nixosModules.mixins-telegraf
-            ];
-          };
-          example-mixins-terminfo = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              dummy
-              fake-hardware
-              self.nixosModules.mixins-terminfo
-            ];
-          };
-          example-mixins-trusted-nix-caches = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              dummy
-              fake-hardware
-              self.nixosModules.mixins-trusted-nix-caches
-            ];
-          };
-          example-mixins-nginx = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              dummy
-              fake-hardware
-              self.nixosModules.mixins-nginx
-            ];
-          };
-
-          # Roles
-          example-roles-github-actions-runner = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              self.nixosModules.roles-github-actions-runner
-              dummy
-              fake-hardware
-              {
-                roles.github-actions-runner.cachix.cacheName = "cache-name";
-                roles.github-actions-runner.cachix.tokenFile = "/run/cachix-token-file";
-                roles.github-actions-runner.tokenFile = "/run/gha-token-file";
-                roles.github-actions-runner.url = "https://fixup";
-              }
-            ];
-          };
-          example-roles-github-actions-runner-github-app = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              self.nixosModules.roles-github-actions-runner
-              dummy
-              fake-hardware
-              {
-                roles.github-actions-runner.cachix.cacheName = "cache-name";
-                roles.github-actions-runner.cachix.tokenFile = "/run/cachix-token-file";
-                roles.github-actions-runner.githubApp = {
-                  id = "1234";
-                  login = "foo";
-                  privateKeyFile = "/run/gha-token-file";
-                };
-                roles.github-actions-runner.url = "https://fixup";
-              }
-            ];
-          };
-          example-roles-nix-remote-builder = nixpkgs.lib.nixosSystem {
-            system = "x86_64-linux";
-            modules = [
-              self.nixosModules.roles-nix-remote-builder
-              dummy
-              fake-hardware
-              {
-                roles.nix-remote-builder.schedulerPublicKeys = [
-                  "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOuiDoBOxgyer8vGcfAIbE6TC4n4jo8lhG9l01iJ0bZz zimbatm@no1"
-                ];
-              }
-            ];
-          };
-        };
+      imports = [
+        ./modules/flake-parts/all-modules.nix
+        ./modules/flake-parts/docs.nix
+      ];
```

Comment:
What if all-modules.nix was also tasked with finding packages in the build tree and mapping them back to packages with a 1:1 mapping?

Then it would look like a flake version of [bld](https://github.com/numtide/bld/blob/main/lib/buildTree.nix)

Reply:
Oh it already does automatically import it. This line ended up there by accident. I will remove it.

Reply:
Sorry, I skimmed your message a bit too quick.

So far it assumes that everything in the tree is a module of some kind.
As long as all packages are declared as flake parts modules under `./modules/flake-parts`, then they will be automatically imported.

We could extend the logic to discover package function files as well. We would either have to come up with a heuristic how to distinguish package functions from modules, or define where package functions can be located.

Personally I prefer to just declare all packages inside flake parts modules or as individual drv-parts modules.